src/components/LocationSearch/LocationSearch.jsx

```diff
@@ -29,42 +36,35 @@ const LocationSearch = ({
   };
 
   return (
-    <div
-      className={`location-search ${
-        isLocationSearchOpen ? "location-search--active" : ""
-      }`}
-    >
-      <button
-        className="close__trigger"
-        onClick={handleLocationSearchVisibility}
-      >
+    <LocationSearchWrapper $isLocationSearchOpen={isLocationSearchOpen}>
```

Comment:
I see you used [transient props](https://styled-components.com/docs/api#transient-props) here. But we should use only with properties that are HTML valid, thinks like `width, color, draggable `, because those would leak to the DOM and can cause some conflicts, but in this case, there's no need, this property exist.